frontend/src/views/auth/Login.vue

```diff
@@ -196,8 +197,14 @@ export default {
       this.error = null
       this.$auth
         .login(this.email, this.password)
-        .then(() => {
+        .then(async () => {
           this.$router.replace(this.$route.query.redirect || '/')
+
+          const user = await this.$auth.loadUser()
+          const profile = await user.profile()._meta.load
+          if (VueI18n.availableLocales.includes(profile.language)) {
+            this.$store.commit('setLanguage', profile.language)
+          }
```

Comment:
The order of the code execution is now:
1. $router.replace is put into the queue
2. loadUser is put into the event queue
3. $router.replace is executed
4. const user gets the user assigned
5. const profile is awaited and assigned
6. language is set

If this order is what should happen, i would await the $router.replace call, that the order of the code execution is clear.

Another way would be to
1. load the user
2. load the profile
3. commit the language
4. replace the route

Like this you don't have pending tasks in the queue when you navigate to the new route.

This is a minor issue and it will work almost always as it is.

Reply:
Good idea, will do